src/contracts/middleware/BLSOperatorStateRetriever.sol

```diff
@@ -0,0 +1,134 @@
+            
+        return operators;
+    }
+
+    /**
+     * @notice returns 
+     *          1) the indices of the quorumBitmaps for each of given operators at the given blocknumber
+     *          2) the indices of the total stakes entries for the given quorums at the given blocknumber
+     *          3) the indices of the stakes of each of the nonsigners in each of the quorums they are a 
+     *             part of (for each nonsigner, an array of length the number of quorums they a part of 
+     *             that are also part of the provided quorumNumbers) at the given blocknumber
+     *          4) the indices of the quorum apks for each of the provided quorums at the given blocknumber
+     * @param referenceBlockNumber is the block number to get the indices for
+     * @param quorumNumbers are the ids of the quorums to get the operator state for
+     * @param nonSignerOperatorIds are the ids of the nonsigning operators
+     */
+    function getCheckSignaturesIndices(
+        uint32 referenceBlockNumber, 
+        bytes calldata quorumNumbers, 
+        bytes32[] calldata nonSignerOperatorIds
+    ) external view returns (CheckSignaturesIndices memory) {
+        uint256 quorumBitmap = BitmapUtils.bytesArrayToBitmap(quorumNumbers);
+
+        CheckSignaturesIndices memory checkSignaturesIndices;
+
+        checkSignaturesIndices.nonSignerQuorumBitmapIndices = registryCoordinator.getQuorumBitmapIndicesByOperatorIdsAtBlockNumber(referenceBlockNumber, nonSignerOperatorIds);
+        checkSignaturesIndices.totalStakeIndices = stakeRegistry.getTotalStakeIndicesByQuorumNumbersAtBlockNumber(referenceBlockNumber, quorumNumbers);
+        
+        checkSignaturesIndices.nonSignerStakeIndices = new uint32[][](nonSignerOperatorIds.length);
+        for (uint i = 0; i < nonSignerOperatorIds.length; i++) {
+            uint192 nonSignerQuorumBitmap = 
+                registryCoordinator.getQuorumBitmapByOperatorIdAtBlockNumberByIndex(
+                    nonSignerOperatorIds[i], 
+                    referenceBlockNumber, 
+                    checkSignaturesIndices.nonSignerQuorumBitmapIndices[i]
+                );
+            // the number of quorums the operator is a part of that are also part of the provided quorumNumbers
+            checkSignaturesIndices.nonSignerStakeIndices[i] = new uint32[](BitmapUtils.countNumOnes(nonSignerQuorumBitmap & quorumBitmap));
+
+            for (uint8 j = 0; j < 192; j++) {
+                // if the operator is a part of the quorum and the quorum is a part of the provided quorums
+                if (nonSignerQuorumBitmap >> j & (quorumBitmap >> j & 1) == 1) {
+                    checkSignaturesIndices.nonSignerStakeIndices[i][j] = stakeRegistry.getStakeUpdateIndexForOperatorIdForQuorumAtBlockNumber(
+                        nonSignerOperatorIds[i],
+                        uint8(j),
+                        referenceBlockNumber
+                    );
+                }
+            }
```

Comment:
So in this case we fill the entry with the proper stake update index, and otherwise (i.e. in all other cases), we leave the value as 0, right?
Possible bug: _what happens when the correct stake update index is zero_? How do we still differentiate?

Reply:
f1faa75d ah, this should fix it. actually was a bug in general compared to how the sigchecker uses it